src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,89 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
 import './FindMovie.scss';
+import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
+import { ResponseError } from '../../types/ReponseError';
+import { MovieCard } from '../MovieCard';
+import { getMovie } from '../../api';
+
+interface Props {
+  movies: Movie[];
+  setMovies: React.Dispatch<React.SetStateAction<Movie[]>>;
+}
+
+export const FindMovie: React.FC<Props> = ({ movies, setMovies }) => {
+  const [queryTitle, setQueryTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [movieFoundError, setMovieFoundError] = useState(false);
+  const [newMovie, setNewMovie] = useState<Movie | null>(null);
+  const [isFindingAgain, setIsFindingAgain] = useState(false);
+  const [movieWasAskedOnce, setMovieWasAskedOnce] = useState(false);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setMovieFoundError(false);
+    setQueryTitle(event.target.value);
+  };
+
+  const handleOnSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+  };
+
+  const handleAddMovieToList = () => {
+    const copyOfMovies: Movie[] = [...movies];
+    const movieAlreadyInList = copyOfMovies.find(movie => {
+      const { imdbId } = newMovie as Movie;
+
+      return movie.imdbId === imdbId;
+    });
+
+    if (!movieAlreadyInList) {
+      setMovies([...copyOfMovies, newMovie as Movie]);
+    }
+
+    setNewMovie(null);
+    setQueryTitle('');
+  };
+
+  const loadMovie = () => {
+    setIsLoading(true);
+
+    getMovie(queryTitle.trim())
+      .catch(() => setMovieFoundError(true))
+      .then(movieFromServer => {
+        const { Response } = movieFromServer as ResponseError;
+
+        if (Response !== 'False') {
+          const {
+            Title, Poster, Plot, imdbID,
+          } = movieFromServer as MovieData;
+
+          const newMovieFromServer: Movie = {
+            title: Title,
+            imgUrl: Poster,
+            description: Plot,
+            imdbId: imdbID,
+            imdbUrl: imdbID,
+          };
+
+          setNewMovie(newMovieFromServer);
+        } else {
+          setMovieFoundError(true);
+          setIsFindingAgain(false);
+          setNewMovie(null);
+        }
+
+        setIsLoading(false);
```

Comment:
you need to move setIsLoading(false) to the finally block because you can get an error  for request and loading will not disappear